apps/barista/src/pages/icon-overview-page/icon-overview-page.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AfterViewInit,
+  Component,
+  OnDestroy,
+  ElementRef,
+  ViewChild,
+  OnInit,
+} from '@angular/core';
+
+import {
+  BaIconOverviewPageContent,
+  BaIconOverviewItem,
+} from '@dynatrace/barista-components/barista-definitions';
+
+import { BaPage } from '../page-outlet';
+import { fromEvent, Subscription } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+
+@Component({
+  selector: 'ba-icon-overview-page',
+  templateUrl: 'icon-overview-page.html',
+  styleUrls: ['icon-overview-page.scss'],
+})
+export class BaIconOverviewPage
+  implements AfterViewInit, OnInit, OnDestroy, BaPage {
+  /** Page contents */
+  contents: BaIconOverviewPageContent;
+
+  /** Filter input field */
+  @ViewChild('filter', { static: true }) _inputEl: ElementRef<HTMLInputElement>;
+
+  /** @internal A list of filtered icons */
+  _filteredIcons: BaIconOverviewItem[];
+
+  /** Subscription on filter change event */
+  private _filterChangeSubscription = Subscription.EMPTY;
+
+  ngOnInit(): void {
+    this._updateFilteredIcons('');
+  }
+
+  ngAfterViewInit(): void {
+    this._filterChangeSubscription = fromEvent(
```

Comment:
You could use banana bracket value bindings with ngModel to get changing value inputs. 
You don't need to create a stream from the input event.
See: https://angular.io/api/forms/NgModel#using-ngmodel-on-a-standalone-control